File: programming/THEDSA/Arrays/reverseArray.cpp
#include<iostream>
using namespace std;
void swap(int *a,int *b){
    int temp=*a;
    *a=*b;
    *b=temp;
}
void reverseArray(int arr[],int n){
    int start=0;
    int end=n-1;
    while(start<end){
        swap(&arr[start],&arr[end]);
        start=start+1;
        end=end-1;
    }
}
int main(){
    int arr[]={60,120,3080,1,3,9,14,100000,0,-1};
    reverseArray(arr,10);
    for(int i=0;i<10;i++){
        cout<<arr[i]<<" ";
        
    }cout<<endl;
}

File: programming/javaClasses/Arrays/ReverseArray.java
import java.util.*;
class ReverseArray{
    public static int[] reverseArray(int arr[]){
        int arrNew[]=new int[arr.length];
        int j=arr.length-1;
        for(int i=0;i<arr.length;i++){
            arrNew[i]=arr[j];
            j--;
        }
        return arrNew;
    }
    public static void swaping(int arr[],int i,int j){
        int temp=arr[i];
        arr[i]=arr[j];
        arr[j]=temp;
    }
    public static int[] reverseArrayOptimised(int arr[]){
        int i=0;
        int j=arr.length-1;
        while(i<j){
            swaping(arr,i,j);
            i++;j--;
        }
        return arr;
    }
    public static void main(String args[]){
        int arr[]={1,2,3,4,5};
        int arrNew[]=reverseArray(arr);
        for(int i=0;i<arrNew.length;i++){
            System.out.print(arrNew[i]+" ");
        }System.out.println();
        reverseArrayOptimised(arrNew);
        for(int i=0;i<arrNew.length;i++){
            System.out.print(arrNew[i]+" ");
        }System.out.println();

    }
}